Data Studio

ranquild opened the pull request:
The Data Studio is the exciting new home for data shaping in metabase:

<img width="1252" height="657" alt="CleanShot 2025-11-20 at 17 13 34" src="https://github.com/user-attachments/assets/1511c80e-71ee-413f-a45d-d3493bc0af82" />


<img width="1250" height="658" alt="CleanShot 2025-11-20 at 17 14 20" src="https://github.com/user-attachments/assets/bc4121c2-cd02-4e30-807e-aaa991bb374b" />
<img width="1265" height="660" alt="CleanShot 2025-11-20 at 17 13 47" src="https://github.com/user-attachments/assets/8cfa002a-88bb-432a-ad56-29e954717505" />


also featuring a new streamlined data picking experience!

<img width="428" height="252" alt="CleanShot 2025-11-20 at 17 15 51" src="https://github.com/user-attachments/assets/15caf3f8-3f84-4268-9c3f-c1ef178bbd9f" />

ranquild reviewed (commented):
@alxnddr the current transform tab hook can be simplified, e.g.
```
const { pathname } = useSelector(getLocation);
if (pathname.startsWith(Urls.transformJobList()) {
  return "jobs";
} 
if (...startsWith(Urls.transformRunList()) {
  return "runs"
}
return "transforms";
```

and you wouldn't hardcode the urls

qnkhuat committed:
tidy things up

wotbrew commented on src/metabase/warehouse_schema/api/table.clj:
Either do something about this, leave a better note or delete the note

wotbrew commented on src/metabase/warehouse_schema/api/table.clj:
Discuss https://metaboat.slack.com/archives/CKZEMT1MJ/p1763034250506099

wotbrew commented on src/metabase/warehouse_schema/api/table.clj:
I would use NOT EXISTS so it can sit with the other predicate terms

uladzimirdev committed:
fix infinite re-render in tree

chodorowicz committed:
test: add bulk sync action e2e test

nvoxland committed:
Merge remote-tracking branch 'origin/master' into workbench2

bpander committed:
Fix some e2e tests from group-9

iethree committed:
fix column selection

bpander committed:
Merge remote-tracking branch 'origin/workbench2' into uxw-2271-minipicker-library

nvoxland committed:
Fixed module config

nvoxland committed:
Merge remote-tracking branch 'origin/workbench2' into workbench2

andsalves committed:
Fix a few e2e tests from oss group

andsalves committed:
Fix e2e tests from groups 46 and 47

andsalves committed:
Fix e2e tests from groups 44 and 45

bpander committed:
Fix group-3 straggler, some of group-8

bpander committed:
Potentially disastrous change (increases item-picker-level of SchemaList and TableList) to fix test in group-8

iethree committed:
do enterprise code correctly

bpander committed:
Fix group-6 straggler, some more of group-8

npfitz committed:
data picker using proper library endpoint. Fixing types I broke. Endless Shame

iethree committed:
library collection search

bpander committed:
Updates some more e2e tests from group-9 (some actual failures left)

bpander committed:
Fix e2e tests from group-14

bpander committed:
Fix e2e tests from group-16

bpander committed:
Fix e2e tests from group-17

andsalves committed:
Fix a couple tests from group 42

bpander committed:
Fix e2e tests from group-18

chodorowicz committed:
tests: add bulk publishing scenario

fix wrong collection link in case of root collection

alxnddr committed:
UXW-2315, UXW-2314

alxnddr committed:
UXW-2294

alxnddr committed:
more UXW-2315, UXW-2314

alxnddr committed:
more UXW-2315, UXW-2314

alxnddr committed:
correct collection types

alxnddr committed:
Revert "Revert "Don't return library collections by default from APIs""

This reverts commit 814c49473b1d6e21d3a0c43490e025151f9891b7.

alxnddr committed:
Revert "Revert "Fixed null type filtering""

This reverts commit 25e0d784082794296303a3263d41f01b35833559.

iethree committed:
Remove Browsing Memory Feature

alxnddr committed:
fetch full tree to have library collections visible

alxnddr committed:
show library in permissions

nvoxland committed:
Fixing tests

nvoxland committed:
Merge remote-tracking branch 'origin/workbench2' into workbench2

uladzimirdev committed:
rewrite unit tests and fix max depth problem

chodorowicz committed:
fix: publish property not validated in the tree view after table publishing

uladzimirdev committed:
Merge branch 'master' into workbench2

qnkhuat committed:
Move bulk editing apis to enterprise (#65968)

kamilmielnik committed:
Make sidebar sticky

chodorowicz committed:
test: bulk attribute actions for tables, schemas and dbs

uladzimirdev committed:
fix more unit tests

kamilmielnik committed:
Copy preview section tests

kamilmielnik committed:
Copy error handling tests

kamilmielnik committed:
Copy undos tests

kamilmielnik committed:
Update test

chodorowicz committed:
test: update new ee endpoints paths

kamilmielnik committed:
Make new bulk table APIs ee-only (fe)
- Closes GDGT-1251

kamilmielnik committed:
Fix global H.DataModel.visit overrides

kamilmielnik committed:
Remove redundant cy.reload

kamilmielnik committed:
Remove redundant cy.reload

uladzimirdev committed:
add select/deselect tests

uladzimirdev committed:
use helper

kamilmielnik committed:
Fix Tree skeleton resizing multiple times during loading
- Closes GDGT-1253

kamilmielnik committed:
Improve icon

kamilmielnik committed:
Fill content

npfitz committed:
don't call getCollectionItems on database collection

uladzimirdev committed:
add search tests and support shift click in search

npfitz committed:
default table publish collection

chodorowicz committed:
fox: reset bulk form values when selection changes

bpander committed:
Fix e2e tests from group-19

chodorowicz committed:
fix:  fix missing dependencies in useEffect gdgt-1245

chodorowicz committed:
test: refactor test setup action order

bpander committed:
Fix e2e tests from group-20

bpander committed:
Fix e2e tests from group-22

bpander committed:
Fix e2e tests from group-23

chodorowicz committed:
test: restore default db as the first step

bpander committed:
Merge remote-tracking branch 'origin/workbench2' into uxw-2271-minipicker-library

# Conflicts:
#	enterprise/frontend/src/metabase-enterprise/data-studio/data-model/components/TablePicker/components/PublishModelsModal.tsx

wotbrew committed:
Kondo the kondo config

npfitz committed:
default collections for metrics and models outside of data studio

wotbrew committed:
Move edit test to ee

wotbrew committed:
Fix list filtering now default is bronze

wotbrew committed:
Move bulk-edit-test to ee

wotbrew committed:
Fix table metadata for test

nvoxland committed:
Revert "Added `collection` argument to search"

This reverts commits:
- 21999e01
- b0f71c03eacab0eca982d7dacb42565408909b96
- 4fa0c3b422afdf1b92106368077e81edf7c5303f

nvoxland committed:
Merge remote-tracking branch 'origin/workbench2' into workbench2

npfitz committed:
constuct initial paths from library collections

nvoxland committed:
Fixed OSS loading of tables

nvoxland committed:
Fixed OSS loading of tables

uladzimirdev committed:
add tests for filtering

andsalves committed:
Merge branch 'workbench2' into uxw-2271-minipicker-library

bpander committed:
Fix e2e tests from group-20

bpander committed:
Fix e2e tests from group-23

andsalves committed:
Fix e2e tests from group 41 / joins-repros

bpander committed:
Fix e2e tests from group-24

bpander committed:
Fix e2e tests from group-26

uladzimirdev committed:
more tests

bpander committed:
Fix e2e tests from group-27

bpander committed:
Fix e2e tests from group-28

bpander committed:
Fix e2e tests from group-30

npfitz committed:
segments editor, hasEnabledNestedQueries

andsalves committed:
Fix a couple more (something between groups 40 and 43)

iethree committed:
keyboard navigation

nvoxland committed:
Fixed module config

nvoxland committed:
Added library/tree API

nvoxland committed:
Added library/tree API

bpander committed:
Fix e2e tests from group-33

andsalves committed:
Fix some e2e tests from groups 40 and 42

bpander committed:
Minor cleanup

bpander committed:
Fix e2e tests from group-34

nvoxland committed:
Fixing tests

npfitz committed:
disable scoped search on the 'database collection', models.cy.spec should be passing now

nvoxland committed:
Fixing tests

andsalves committed:
Fix e2e tests from group 37 (except one depending on Recents tab)

bpander committed:
Starts fixing e2e tests from oss group

iethree committed:
cleanup

iethree committed:
wait to focus

alxnddr committed:
hide our analytics in the data studio

alxnddr committed:
merge library and data studio

alxnddr committed:
change backend library flag validation to data-studio

kamilmielnik committed:
Merge branch 'master' into workbench2

kamilmielnik committed:
Make bulk edit inputs non-clearable
- This prevents sending "empty" requests when clearing values
- And clearing does not really make sense here

chodorowicz committed:
fix: test: table rescan and discard mocks are wrong

chodorowicz committed:
fix: test: wrong mock in oss sync table schema

wotbrew committed:
Remove ^:parallel from bulk edit tests

They can contend and cause deadlocks

chodorowicz committed:
fix: test: failure due to repeated mocking of same route

chodorowicz committed:
test: refactor: cypress table tests

kamilmielnik committed:
Update e2e test

kamilmielnik committed:
Update e2e test

kamilmielnik committed:
Update e2e test

uladzimirdev committed:
fix browse nav section test

kamilmielnik committed:
Add missing token

uladzimirdev committed:
add check icon for published tables

nvoxland committed:
Removed unused import

bpander committed:
Updates some e2e tests from oss group, removes some now-invalid tests

chodorowicz committed:
test: use default external db, skip transform test

bpander committed:
Starts fixing e2e tests from mongo group

uladzimirdev committed:
remove background from not active table in search

alxnddr committed:
specs

bpander committed:
Fix e2e tests from group-15

alxnddr committed:
hack fix ci while i'm on meetings

npfitz committed:
scoped search is back

nvoxland committed:
Merge remote-tracking branch 'origin/master' into workbench2

# Conflicts:
#	resources/migrations/058_update_migrations.yaml

bpander committed:
Fix some e2e tests from group-32

npfitz committed:
table picker fixes

nvoxland committed:
Fixed merge

bpander committed:
Fix e2e test from group-42

alxnddr committed:
Merge remote-tracking branch 'origin/master' into workbench2

iethree committed:
Merge branch 'workbench2' into uxw-2271-minipicker-library

# Conflicts:
#	enterprise/frontend/src/metabase-enterprise/api/tags.ts
#	enterprise/frontend/src/metabase-enterprise/data-studio/index.ts
#	enterprise/frontend/src/metabase-enterprise/data-studio/utils.ts
#	enterprise/frontend/src/metabase-enterprise/library/index.ts
#	frontend/src/metabase/lib/icon.ts
#	frontend/src/metabase/plugins/index.ts

iethree committed:
merge fixes

iethree committed:
fix embedding types

npfitz committed:
re-enable the recents tab on the data picker modal

nvoxland committed:
Avoid race condition on list-table-filtering-test

iethree committed:
mini picker unit tests

nvoxland committed:
Fixing tests

alxnddr committed:
ci, fixes

iethree committed:
fix library collection bugs

iethree committed:
cleanup data studio plugin

iethree committed:
fix lots of unit tests

alxnddr committed:
Merge remote-tracking branch 'origin/master' into workbench2

iethree committed:
Merge branch 'uxw-2271-minipicker-library' into workbench2

alxnddr committed:
fix import

alxnddr committed:
fix

nvoxland committed:
Fixed unused import

nvoxland committed:
Fixed accidental #p

alxnddr committed:
ci

alxnddr committed:
revert util

alxnddr committed:
specs

alxnddr committed:
specs

alxnddr committed:
specs

sloansparger committed:
fix bug / bad type in SchemaFormSelect (#65995)

kamilmielnik committed:
Activate token after restore

wotbrew committed:
Use actual anti join in GET /table query

wotbrew committed:
Remove old comments

chodorowicz committed:
test: unskip table metadata transform test

chodorowicz committed:
test: fix search tables e2e test

chodorowicz committed:
fix: test: selection in table picker

kamilmielnik committed:
Fix Source input disabled when filtering by Metabase transform
- Closes GDGT-1259

chodorowicz committed:
test: fix: filter unused table

kamilmielnik committed:
Fix e2e test

chodorowicz committed:
fix: test: datamodel toasts and errors

kamilmielnik committed:
Fix e2e test

bpander committed:
Fix e2e tests from group-9

kamilmielnik committed:
Fix flaky tests

stasgavrylov committed:
GDGT-1244: Fix DB tree virtualization (#66125)

* GDGT-1244: Fix tree virtualization. Replace top with translateY to improve performance

* GDGT-1244: Fix scroll bug. Refactor border styling

bpander committed:
Fix e2e test from group-22

alxnddr committed:
bring back modeling empty state

chodorowicz committed:
test: adjust publishing test

bpander committed:
Fix e2e test from group-25

iethree committed:
fix allowing cross DB joins

bpander committed:
Fix e2e test from group-26

edpaget commented on src/metabase/collections/models/collection.clj:
nit if these are boolean functions they should be `check-allowed-content?` since I think these raise in the enterprise version if content is not allowed maybe they could just take the model as an argument and return the model.

edpaget commented on enterprise/backend/src/metabase_enterprise/library/api.clj:
Maybe it doesn't need to error it could just return 204?

edpaget commented on enterprise/backend/src/metabase_enterprise/library/api.clj:
the REST nerd in me needs to say this should be a PUT since it (can be idempotent), but that isn't important obviously

bpander committed:
Fix straggler from oss group

edpaget commented on test/metabase/warehouse_schema/models/table_test.clj:
```suggestion
                   :model/Card  {c1 :id} {:type :model :published_table_id t1}
                   :model/Card  _ {:type :model :published_table_id t1 :archived true}
                   :model/Card  {c3 :id} {:type :model :published_table_id t1}
                   :model/Card  _ {:type :model :published_table_id t2 :archived_directly true}
                   :model/Card  _ {:type :metric :published_table_id t2}]
```

edpaget reviewed (commented):
Two big points:
* delete commented out test code
* have response schemas were needed for new API routes (metabot flagged these mostly)

On a style point I think the check-allowed-* functions should take a model and return the model throwing if there is a validation error. This makes it easier to use them in threading macros

iethree committed:
properly disable invalid items in MiniPicker

bpander committed:
Starts fixing e2e tests from group-32

johnswanson commented on src/metabase/models/interface.clj:
I think `nil` is a valid `:any`

npfitz committed:
unit test adjustments

andsalves committed:
Fix remaining issue with e2e group 37

iethree committed:
fix item filtering

alxnddr committed:
Merge branch 'master' into workbench2

iethree committed:
add recent views to mini picker

github-automation-metabase committed:
Update Loki Snapshots

johnswanson committed:
Fix flaky Remote Sync test (#65862)

iethree committed:
test recents and backparsing

nemanjaglumac committed:
Clean up backend-test matrix (#66158)

iethree committed:
additional filtering test

alxnddr committed:
fix minipicker shows items it should not show

nvoxland committed:
Fixed :here in library API

nvoxland committed:
Merge remote-tracking branch 'origin/workbench2' into workbench2

nvoxland committed:
Fixed :here in library API

npfitz committed:
UXW-2358 publishing outside of the data studio

nvoxland committed:
Fixed :here in library API

nvoxland committed:
Merge remote-tracking branch 'origin/workbench2' into workbench2

alxnddr committed:
snapshot

bpander committed:
Fix most e2e tests from group-32

deniskaber committed:
Fix timeseries value detection (#66161)

nvoxland committed:
Lint fixes

nvoxland committed:
Fixed failing test

dpsutton committed:
Prevent driver tests (#66156)

Co-authored-by: Bryan Maass <bryan.maass@gmail.com>

alxnddr committed:
revert fix

nvoxland committed:
Reapply "Added `collection` argument to search"

This reverts commit f28d9d5af928495bfca08414ee03feb68fcd8588.

nvoxland committed:
Merge remote-tracking branch 'origin/workbench2' into workbench2

nemanjaglumac committed:
Fetch uberjar using a proper action (#66165)

andsalves committed:
Quick fix in CollectionPicker/utils.ts

iethree committed:
fix embedding column picking

andsalves committed:
Merge branch 'master' into workbench2

nvoxland committed:
Trying CI with search_index.verified

nvoxland committed:
Fixed failing test

kamilmielnik committed:
Translate label

nvoxland committed:
Fixed failing test

nvoxland committed:
Merge remote-tracking branch 'origin/workbench2' into workbench2

chodorowicz committed:
fix: published info not updated in the tree picker after deleting model

chodorowicz committed:
fix: refactor: publish collection link tests

nvoxland committed:
Fixed failing test

nvoxland committed:
Merge remote-tracking branch 'origin/workbench2' into workbench2

bpander committed:
Fix remaining e2e tests from group-32

npfitz committed:
FileUploadStatus unit test, quesiton reproductions

npfitz committed:
create data studio plugin file

iethree committed:
fix hidden items some more

alxnddr committed:
fix dark mode

bpander committed:
Fix issue where confirmation modal closes then reopens again then closes again when you archive a dashboard

npfitz committed:
happy types

alxnddr committed:
UXW-2367

npfitz committed:
resolving more merge sadness

alxnddr committed:
UXW-2378

npfitz committed:
Merge remote-tracking branch 'origin/workbench2' into workbench2

nvoxland committed:
Fixed merge issues

alxnddr committed:
this element is not visible because you can't even open incompatible sample database db

bpander committed:
Fix e2e test from group-14

bpander committed:
UXW-2279 Changes Data Studio's Glossary page background color to background-secondary

iethree committed:
fix embedding data picker clicks

iethree committed:
filtering fixes for transforms

alxnddr committed:
UXW-2372

alxnddr committed:
update mini picker specs after a fix

iethree committed:
fix flaky join test

nvoxland committed:
Fixing in-place support for collections

chodorowicz committed:
fix: cannot scroll search results

chodorowicz committed:
ui: tweak search results checkbox spacing to match tree results

chodorowicz committed:
fix: scroll position jump when entering field sort mode gdgt-1269

chodorowicz committed:
Revert "fix: scroll position jump when entering field sort mode gdgt-1269"

This reverts commit bfecdec6612e391e6e13ed2f23b62fb5a449c4c9.

npfitz commented on enterprise/frontend/src/metabase-enterprise/data-studio/app/pages/ModelingSectionLayout/ModelingSidebar/LibrarySection/CreateLibraryModal/CreateLibraryModal.tsx:
See Modal Comment

nvoxland commented on enterprise/backend/src/metabase_enterprise/library/api.clj:
Had it as a POST since we tend to use PUT for updating and this creates it

nvoxland commented on enterprise/backend/src/metabase_enterprise/library/api.clj:
I figured a 400 would make more since becase it's not doing what you are expecting it to and not going to return what you expect. The FE shouldn't be calling this if the library exists ever, so it's more of a fallback catch for it doing something weird so good to make that more explicit.

alxnddr reviewed (approved):
lgtm

npfitz reviewed (commented):
Just a couple comments, nothing major
